src/wrapped_radix_engine_toolkit/low_level/transaction_manifest/instructions/instructions.rs

```diff
@@ -0,0 +1,306 @@
+use crate::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, derive_more::Display, uniffi::Record)]
+#[display("{}", self.instructions_string())]
+pub struct Instructions {
+    pub(crate) secret_magic: InstructionsSecretMagic, // MUST be first prop, else you break build.
+    pub network_id: NetworkID,
+}
+
+impl Deref for Instructions {
+    type Target = Vec<ScryptoInstruction>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.secret_magic.0
+    }
+}
+
+impl Instructions {
+    pub(crate) fn from_scrypto(
+        instructions: ScryptoInstructions,
+        network_id: NetworkID,
+    ) -> Self {
+        Self {
+            secret_magic: instructions.into(),
+            network_id,
+        }
+    }
+}
+
+impl Instructions {
+    pub fn instructions_string(&self) -> String {
+        let network_definition = self.network_id.network_definition();
+        scrypto_decompile(self, &network_definition).expect("Should never fail, because should never have allowed invalid instructions")
```

Comment:
@0xOmarA safe to unwrap `decompile` on `&[InstructionV1]` **given network is correct**? Or when can it fail?